templates/components/form/item_device.html.twig

```diff
@@ -190,7 +190,7 @@
                         __('Status'),
                         field_options|merge({
                            'entity': item.fields['entities_id'],
-                           'condition': item.getStateVisibilityCriteria() ?? {}
+                           'condition': item is usingtrait('Glpi\\Features\\State') ? item.getStateVisibilityCriteria() : []
```

Comment:
This condition shoule be moved to hide the full field. But an item with a `states_id` that does not use new feature is an error that should be handled in tests.

Reply:
Maybe the condition could be `{% if item.isField('states_id') and item.getType() in config('state_types') %}`. The `tests\units\State::testHasFeature()` almost already checks that items from `state_types` are using the trait. Test could be enhanced a bit to really check the trait itself.